src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -176,8 +174,20 @@ internal ItemManager(RecentItemsManager recentItemsManager)
 
             // Use a monotonically increasing integer to keep track the original alphabetical order of each item.
             var currentIndex = 0;
+
             var builder = ArrayBuilder<MatchResult>.GetInstance();
+            if (KeepAllItemsInTheList(initialRoslynTriggerKind, filterText) && !needToFilterExpanded && !needToFilter)
+            {
+                // PERF: Conditionally set the capacity of the ArrayBuilder, i.e. if we think it's likely that the filtering result
+                // would be similar in size to the initial list, in which case, setting the capacity would avoid call to Resize later.
+                // Otherwise, don't set capacity as an attempt to avoid allocating large array which might unnecessarily end up in LOH.
+                // We say it's likely that we end up with a list of similar size after filtering if:
+                // 1. We need to show items even if they don't match filter text
+                // 2. No filter is selected
+                builder.EnsureCapacity(data.InitialSortedList.Length);
+            }
```

Comment:
was this a speculative perf change?  or was this driven from some actual data?  i have no problem with it either way, i'm just curious.

Reply:
This is addressing RPS regression caught in validation PR.

Before: +6.3mb LOH allocation
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS/pullrequest/278169

After: +610kb  LOH allocation
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS/pullrequest/278694

Reply:
@sharwell  is your sdgmented array ready?

Reply:
That won't help right now, since ImmutableArray is part of VS API.